packages/sdk/src/swap/sdk.ts

```diff
@@ -400,4 +402,135 @@ export class SwapSDK {
       ...getAssetAndChain(depth.asset),
     }));
   }
+
+  async requestDepositAddressV2({
+    quote,
+    destAddress,
+    fillOrKillParams: inputFoKParams,
+    affiliateBrokers: affiliates,
+    ccmParams,
+    brokerCommissionBps,
+  }: DepositAddressRequestV2) {
+    await this.validateSwapAmount(quote.srcAsset, BigInt(quote.depositAmount));
+    assert(quote.type === 'DCA' || quote.type === 'REGULAR', 'Invalid quote type');
```

Comment:
should we  throw if `type === 'dca'` but `!quote.dcaParams`

Reply:
was there, but i factored it out into a common helper